mirgecom/steppers.py

```diff
@@ -27,8 +27,35 @@
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 """
+
+import numpy as np
 from logpyle import set_dt
 from mirgecom.logging_quantities import set_sim_state
+from pytools import memoize_in
+
+
+def compile_timestepper(actx, timestepper, state, rhs):
+    """Create lazy evaluation version of the timestepper."""
+    @memoize_in(actx, ("mirgecom_compiled_operator",
+                       timestepper, rhs,
+                       tuple(len(el) for el in state)))
+    def get():
+        return actx.compile(lambda y, t, dt: timestepper(state=y, t=t,
+                                                         dt=dt,
+                                                         rhs=rhs),
+                            [state, np.float64(0), np.float64(0)])
```

Comment:
```suggestion
        return actx.compile(lambda y, t, dt: timestepper(state=y, t=t,
                                                         dt=dt,
                                                         rhs=rhs))
```

Reply:
I am curious about this; dt is sometimes constant, but sometimes a function. Do we need to account for that?  If dt gets fixed in time - then this can be the reason that the simulation steps beyond t_final.  Also, if `dt` is not allowed to change during the simulation, we have a problem.

Reply:
The compiled function will take 3 arguments, one for each `y, t, dt`. So, chaging `dt` won't re-trigger a compilation.

Reply:
Lambda noob